cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CopyNativeLibTask.java

```diff
@@ -239,6 +243,42 @@ private void copyLibsFromBalo(Path baloFilePath, Set<Path> moduleDependencySet)
         }
     }
 
+    private void copyAndValidateBaloDependencies(BuildContext buildContext, Path importDependencyPath,
```

Comment:
If I import 2 modules (mod1, mod2) and mod1 has compiler dependency and same compiler dependency add as default in mod2. But that dependency will be packed in mod2 balo and at runtime everything will work fine. But still warning message will be shown during compilation as per below logic. Shall we fix that case as well

Reply:
Still, we need that warning message as dependencies of mod2 will be copied to the  mod2 and the current module only. For compiling mod1, we need to copy that dependency to mod1. As we are copying the dependencies based on the modules, the user have to define that dependency for mod1. Please correct me if I'm wrong.